File: problems/contests/dated/2020_06_12/4_hdu4578/gen.py
MXN = 100000
MXM = 100000
MXC = 10000

MXT = 1
MXP = 1

MXC = 10
MXN = 5
MXM = 5

from random import randint
T = randint(1, MXT)
for i in range(T):
    N = randint(1, MXN)
    M = randint(1, MXM)
    print(N, M)
    for j in range(M):
        l = randint(1, N)
        r = randint(l, N)
        if (randint(0, 1)):
            print(4, l, r, randint(1, MXP))
        else:
            print(randint(1, 3), l, r, randint(1, MXC))
print('0 0')

File: problems/uva/x_10817down/gen.cpp
#include <ctime>
#include <cstdio>
#include <cstdlib>

//const int MXT = 10;
//const int MXS = 8;
//const int MXM = 20;
//const int MXN = 100;
//const int MXC = 40000;

const int MXT = 100;
const int MXS = 8;
const int MXM = 20;
const int MXN = 100;
const int MXC = 40000;

int main()
{
    srand(time(0));
    for (int t=0; t<MXT; ++t)
    {
        int S = rand()%MXS+1;
        int M = rand()%MXM+1;
        int N = rand()%MXN+1;
        printf("%d %d %d\n", S, M, N);
        for (int i=0; i<M+N; ++i)
        {
            //printf("%d", rand()%MXC +1e4+1);
            printf("%d", rand()%MXC+1);
            for (int i=1; i<=S; ++i) if (rand()%2)
                printf(" %d", i);
            printf("\n");
        }
    }
    printf("0 0 0\n");
}

